### PR TITLE
Fix time granularity-related issues

### PR DESCRIPTION
**Pull request by mistercrunch**
related to https://github.com/apache/incubator-superset/pull/4755

@betodealmeida 👀

### CODE REVIEW

**Comment by betodealmeida** [comment]
Taking a look — thanks for fixing this!

File: superset/models/core.py [hunk 1]
```diff
@@ -786,7 +786,12 @@ def grains(self):
         return self.db_engine_spec.time_grains
 
     def grains_dict(self):
-        return {grain.duration: grain for grain in self.grains()}
+        """Allowing to lookup grain by either label or duration
+
+        For backward compatibility"""
+        d = {grain.duration: grain for grain in self.grains()}
+        d.update({grain.label: grain for grain in self.grains()})
```
Comment (betodealmeida) [inline superset/models/core.py]: How do the grains show up in the form? Do we have duplicates?

Comment (mistercrunch) [reply superset/models/core.py]: Looks ok. The dropdown uses the `grains` property, not `grains_dict`.

Comment (betodealmeida) [reply superset/models/core.py]: Awesome, thanks!

**Review by betodealmeida — APPROVED** [review]

**Pull request merged**